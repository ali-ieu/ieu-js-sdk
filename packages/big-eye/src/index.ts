import initBrowserLogger from './thirdParty/logger/index'
let _envFlag = 'development'
let BrowserLogger: any

function getImgNaturalDimensions(img: HTMLImageElement): number[] {
    let nWidth = 0,
        nHeight = 0
    if (img.naturalWidth) {
        nWidth = img.naturalWidth
        nHeight = img.naturalHeight
    }
    return [nWidth, nHeight]
}

const mutationCallback: MutationCallback = function (mutationsList) {
    mutationsList.forEach(function (mutation) {
        mutation.addedNodes.forEach(function (node) {
            dfs(node, function (node) {
                function onLoadImage(): void
                function onLoadImage(img?: HTMLImageElement): void {
                    const params = img === undefined ? (node as HTMLImageElement) : img
                    const [nWidth, nHeight] = getImgNaturalDimensions(params)
                    const minLen = Math.min(nWidth, nHeight)
                    // 2倍屏设计稿750x1334
                    if (minLen > 750) {
                        if (_envFlag === 'development') {
                            // 颠倒和高亮
                            const _node = node as HTMLElement
                            _node.style.transform = 'rotate(180deg)'
                            _node.style.backgroundColor = 'royalblue'
                        } else {
                            // 上报
                            const imgSrc = img === undefined ? (node as HTMLMediaElement).currentSrc : img.currentSrc
                            BrowserLogger.avg(imgSrc, minLen)
                        }
                    }
                }
                // node.currentStyle 兼容 IE、FireFox
                const style = (node as any).currentStyle || window.getComputedStyle(node as HTMLElement)
                if (
                    (node as HTMLElement).attributes &&
                    ((node as HTMLElement).attributes as any).mmtrust &&
                    ((node as HTMLElement).attributes as any).mmtrust.nodeValue === 'true'
                ) {
                    return
                }
                if ((node as HTMLElement).tagName === 'IMG') {
                    node.addEventListener('load', onLoadImage, false)
                } else if (style.backgroundImage != 'none') {
                    const img = new Image()
                    img.addEventListener('load', onLoadImage.bind(null, img), false)
                    img.src = style.backgroundImage.slice(4, -1).replace(/"/g, '')
                }
            })
        })
    })
}

function dfs(parentNode: any, cb: (node: Node) => void) {
    if (!parentNode) return
    cb(parentNode)
    console.log(parentNode)
    if (parentNode.children) {
        Array.prototype.forEach.call(parentNode.children, function (child) {
            dfs(child, cb)
        })
    }
}

function monitorInit(target: HTMLElement, config?: MutationObserverInit) {
    if (!window.MutationObserver) return
    config = {
        ...(config || {}),
        ...{ attributes: true, childList: true, subtree: true },
    }

    const observer = new MutationObserver(mutationCallback)
    observer.observe(target, config)
}

function BigEye({ targetId, envFlag, loggerConfig }: IEntry) {
    const targetObj = document.getElementById(targetId || 'app')
    envFlag && (_envFlag = envFlag)

    BrowserLogger = initBrowserLogger({
        env: envFlag,
        ...loggerConfig,
    })
    if (targetObj) {
        document.addEventListener('DOMContentLoaded', function () {
            monitorInit(targetObj)
        })
    }
}

export { BigEye, BrowserLogger }
