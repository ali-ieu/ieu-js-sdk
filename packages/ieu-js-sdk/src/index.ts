export { EjoySDK } from '@ali-ieu/ejoysdk'
export { ThirdParty } from '@ali-ieu/third-party'
