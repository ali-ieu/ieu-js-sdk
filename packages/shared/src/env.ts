import { logError } from '.'

interface EnvOption {
    host: string
    /** 使用裂变账号服务地址 */
    fissionURL?: string
}
class Env {
    private host: string
    private fissionURL?: string
    constructor(options: EnvOption) {
        this.host = this.completeHost(options.host)
        this.fissionURL = options.fissionURL
    }

    private completeHost(host: string) {
        if (!host) {
            return host
        }
        if (!/^https?/.test(host)) {
            return `https://` + host
        }
        return host
    }

    setHost(host: string): void {
        this.host = this.completeHost(host)
    }

    getHost(): string {
        if (!this.host) {
            if (__DEV__) {
                logError('[host] 尚未初始化，请按照文档指示，设置一个明确的 Host 值!')
            }
        }
        return this.host
    }

    getFissionURL(): string {
        if (!this.fissionURL) {
            if (__DEV__) {
                logError('[fissionURL] 尚未初始化，请按照文档指示，设置一个明确的 fissionURL 值!')
            }
        }
        return this.fissionURL ?? ''
    }
}

export { Env }
