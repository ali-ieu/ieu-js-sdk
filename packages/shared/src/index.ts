export * from './env'
export * from './store'
export * from './utils'
// not support yet
// export * as Utils from './utils'
