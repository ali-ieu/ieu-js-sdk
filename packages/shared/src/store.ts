import { logError } from '.'

// localStorage may be unavailable in webview
function check() {
    if (__DEV__ && typeof window.localStorage === null) {
        console.error('window.localStorage is unavailable')
    }
}

type ExpiryData = {
    value: any
    expiry: number
    version: string
}

interface StoreOptions {
    version: string
    nameSpace: string
}
class Store {
    private nameSpace: string
    private version: string
    private store: Window['localStorage']
    constructor(options: StoreOptions) {
        check()
        this.nameSpace = options.nameSpace
        this.version = options.version
        this.store = window.localStorage
    }

    private getKey(key: string): string {
        if (!this.nameSpace) {
            if (__DEV__) {
                logError('[nameSpace] 尚未初始化')
            }
        }
        return this.nameSpace + ':' + key
    }

    setNameSpace(nameSpace: string): void {
        this.nameSpace = nameSpace
    }

    /**
     * expiry: 时间戳，过期时间
     */
    setItem(key: string, value: string | number | boolean | Record<string, any>, expiry: number): void {
        key = this.getKey(key)
        const expiryData: ExpiryData = {
            value,
            expiry,
            version: this.version,
        }
        this.store.setItem(key, JSON.stringify(expiryData))
    }

    getItem<T = any>(key: string): T | null {
        key = this.getKey(key)
        let result: ExpiryData | null = null
        try {
            result = JSON.parse(this.store.getItem(key) as string)
            if (result) {
                if (result.version === this.version) {
                    if (Date.now() < result.expiry) {
                        return result.value
                    } else {
                        // 失效移除
                        this.removeItem(key)
                        return null
                    }
                }
                return null
            }
        } catch (e: unknown) {
            console.error(e)
        }
        return null
    }

    removeItem(key: string): void {
        key = this.getKey(key)
        this.store.removeItem(key)
    }

    clear(): void {
        this.store.clear()
    }
}

export { Store }
