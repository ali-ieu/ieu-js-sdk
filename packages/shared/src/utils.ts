function isNative(): boolean {
    return /HYSDK/gi.test(navigator.userAgent) || /** 兼容客户端某些异常场景 */ Boolean(window['webkit'])
}

function logError(errorMsg: string, throwInDev = true): void {
    if (throwInDev) {
        throw new Error(errorMsg)
    } else {
        console.error(errorMsg)
    }
}
export { isNative, logError }
