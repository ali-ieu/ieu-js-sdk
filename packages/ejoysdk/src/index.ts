import { isNative } from '@ali-ieu/shared'
import { EventEmitter } from 'events'

type NativeResult = {
    error?: any
    /** 正常时 */
    retval?: any
}

interface Params {
    /** method name */
    name: string
    args?: any
}

interface BaseConfig {
    share_support_platform: {
        [k in ShareType]?: boolean
    }
}

export interface NativeShareParams {
    /** 标题 */
    title?: string
    message?: string
    /** 分享链接地址 */
    content_url?: string
    media?: ShareMedia[]
}

type ShareMedia = {
    /** image_url => 图片路径, image_data => 图片 base64 */
    type: 'image_url' | 'image_data'
    // type: LiteralUnion<'image_url' | 'image_data', string>;
    data: string
}

type BaseShareType = 'timeline' | 'messenger'
type Channel = 'wechat' | 'facebook'

/** 支持的分享类型 */
export type ShareType = `${Channel}_share_${BaseShareType}` | 'qq_share_messenger' | 'qq_share_qzone' | 'sina_share_weibo'

export type ShareResult = {
    /** 0 表示分享成功，其他表示失败 */
    code: number
    /** 分享状态相关描述 */
    msg: string
}

/**
 * 平台分享的返回体
 */
export type ShareSupportInfo =
    | boolean
    | {
          /** --用户手机上是否有安装对应平台的客户端 */
          has_installed: boolean
          /** --游戏是否有配置对应平台的分享参数 */
          has_share_infos: boolean
          /** --友盟sdk是否支持对应平台分享 */
          is_support: boolean
      }

/**
 * EjoySDK 实例
 *
 */
class SDK {
    messager: EventEmitter
    eventsId: string

    constructor() {
        if (!window.ejoysdk) {
            // Webview 通过触发此回调来传递消息
            window.ejoysdk = {
                nativeCallback: this.nativeCallback,
            }
        }
        this.messager = new EventEmitter()
        this.eventsId = '1000'
    }

    private genID(): string {
        this.eventsId = String(Number(this.eventsId) + 1)
        return this.eventsId
    }

    private getPhoneType() {
        const ua = navigator.userAgent
        const isIOS = /iPad|iPhone|iPod/i.test(navigator.platform) || (navigator.platform === 'MacIntel' && navigator.maxTouchPoints > 1)
        if (/android/i.test(ua)) {
            return 'android'
        } else if (isIOS) {
            return 'ios'
        }
        return 'other'
    }

    private native(params: Params, callback: (err: Error | null, result?: any) => void) {
        const eventsId = this.genID()
        this.messager.once(eventsId, (result: NativeResult) => {
            // 注册 分发回去
            if (result.error) {
                callback(new Error(result.error))
                return
            }
            try {
                // 如果是可以反序列化的字符串，就反序列化回去
                callback(null, JSON.parse(result.retval))
            } catch (e) {
                // 否则，原样返回
                callback(null, result.retval)
            }
        })
        // 不在支持的客户端环境
        if (!isNative()) {
            this.nativeCallback(eventsId, { error: '不被支持的 webview 环境' })
            return
        }

        const type = this.getPhoneType()
        switch (type) {
            case 'android':
                {
                    const defaultValue = JSON.stringify({
                        id: eventsId,
                        name: params.name,
                        args: params.args,
                        asyncCall: true,
                    })
                    if (params.name === 'notifyLua') {
                        // 安卓事件异步发送
                        prompt('jsinterface://', defaultValue)
                    } else {
                        // 使用同步阻塞的方式拿到结果
                        const resultText = prompt('jsinterface://', defaultValue)
                        const result = resultText ? JSON.parse(resultText) : {}
                        this.nativeCallback(eventsId, result)
                    }
                }
                break
            case 'ios':
                // ios 事件发送
                {
                    const query = {
                        id: eventsId,
                        name: params.name,
                        args: params.args,
                        asyncCall: params.name === 'notifyLua',
                    }
                    const queryString = encodeURIComponent(JSON.stringify(query))

                    const webViewJSinterface = window?.['webkit']?.['messageHandlers']?.['jsinterface']

                    if (webViewJSinterface) {
                        // WKWebView, iOS 8.0+
                        webViewJSinterface.postMessage(queryString)
                    } else {
                        // UIWebView, iOS 8.0-
                        let iframe: HTMLIFrameElement | null = document.createElement('iframe')
                        iframe.style.display = 'none'
                        iframe.src = `jsinterface:///?${queryString}`
                        document.documentElement.appendChild(iframe)
                        iframe.parentNode && iframe.parentNode.removeChild(iframe)
                        iframe = null
                    }
                }
                break
            default:
                // 不被支持的方式
                this.nativeCallback(eventsId, { error: '当前设备不支持' })
                break
        }
    }

    nativeCallback = (id: string, result: NativeResult): void => {
        this.messager.emit(id, result)
    }

    /**
     * 获取应用初始化的信息
     */
    getStartupData(_: Record<string, any>, callback: (err: Error | null, result?: BaseConfig) => void): void {
        this.native({ name: 'getStartupData', args: {} }, callback)
    }

    /** 调用端内分享功能 */
    share(shareType: ShareType, shareParams: NativeShareParams, callback: (err: Error | null, result?: ShareResult) => void): void {
        this.native(
            {
                name: 'notifyLua',
                args: {
                    chl: 'social',
                    type: shareType,
                    params: shareParams,
                },
            },
            callback,
        )
    }

    /** 检查某种分享方式是否被支持 */
    checkShareType(types: ShareType[], callback: (err: Error | null, result?: ShareSupportInfo[]) => void): void {
        this.native(
            {
                name: 'notifyLua',
                args: {
                    chl: 'lua_call',
                    syncCall: true,
                    params: { module: 'ejoysdk_lua.social.ejoysdk_social', func: 'is_support', params: types },
                },
            },
            callback,
        )
    }

    /** 设置 webview 透明背景 */
    transparentBackground(_: Record<string, any>, callback: (err: Error | null, result?: any) => void): void {
        this.native(
            {
                name: 'transparentBackground',
                args: {},
            },
            callback,
        )
    }

    /** 关闭 webview */
    closeWebview(_: Record<string, any>, callback: (err: Error | null, result?: any) => void): void {
        this.native(
            {
                name: 'closeBrowser',
                args: {},
            },
            callback,
        )
    }

    /** 是否端内外 */
    isAliHYApp(): boolean {
        let isAliHYApp = false
        const myUA = navigator.userAgent
        if (myUA && myUA.indexOf('HYSDK') > 0) {
            isAliHYApp = true
        }
        if (window['webkit'] && window['webkit']['messageHandlers'] && window['webkit']['messageHandlers']['jsinterface']) {
            // 此为ios特有，如果存在此对象，可以主动判断为端内
            isAliHYApp = true
        }
        return isAliHYApp
    }
}

// 单实例
const EjoySDK = new SDK()

export { EjoySDK }
