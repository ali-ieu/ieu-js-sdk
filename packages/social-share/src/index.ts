import { isNative, logError } from '@ali-ieu/shared'
import { EjoySDK, ShareResult, NativeShareParams } from '@ali-ieu/ejoysdk'
import { EventEmitter } from 'events'

interface SocialShareConfig {
    platform?: 'web' | 'native'
    wxConfig?: {
        debug?: boolean
        appId: string
        timestamp: number
        nonceStr: string
        signature: string
        jsApiList: string[]
    }
}

export interface ShareParams {
    title?: string
    message?: string
    link?: string
    /** base64 或者 网络资源地址 */
    medias?: string[]
}

interface FaceBookShareParams extends ShareParams {
    type?: 'messenger' | 'timeline'
}

interface TwitterShareParams extends ShareParams {
    type?: 'messenger' | 'timeline'
}

interface QQShareParams extends ShareParams {
    type?: 'messenger' | 'qzone'
}

interface SinaShareParams extends ShareParams {
    type?: 'weibo'
}

interface WechatShareParams extends ShareParams {
    type?: 'messenger' | 'timeline'
}

class SocialShare {
    private platform: 'web' | 'native'
    private wx: any
    private messager?: EventEmitter

    constructor(config: SocialShareConfig) {
        this.platform = config.platform || (isNative() ? 'native' : 'web')
        if (config.wxConfig) {
            // 配置 wechat
            this.messager = new EventEmitter()
            this.messager.on('wechatShare', (fn) => {
                this._initWxSDK().then(() => {
                    this.wx.config(config.wxConfig)
                    this.wx.ready(() => {
                        fn()
                    })
                })
            })
        }
    }

    private _initWxSDK() {
        return new Promise((resolve: any) => {
            const setWX = () => {
                const oScript = document.createElement('script')
                oScript.type = 'text/javascript'
                oScript.onload = () => {
                    this.wx = window.wx
                    resolve()
                }
                oScript.src = 'https://res.wx.qq.com/open/js/jweixin-1.4.0.js'
                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                document.querySelector('head')!.appendChild(oScript)
            }

            if (this.wx) {
                resolve()
            } else if (window.wx) {
                this.wx = window.wx
                resolve()
            } else {
                setWX()
            }
        })
    }

    private convertParams(params: ShareParams): NativeShareParams {
        const result: NativeShareParams = {}
        if (typeof params.link !== 'undefined') {
            result.content_url = params.link
        }
        if (typeof params.medias !== 'undefined') {
            result.media = params.medias.map((c) => {
                if (/^http:?/.test(c)) {
                    return {
                        type: 'image_url',
                        data: c,
                    }
                } else {
                    // as Base64
                    return {
                        type: 'image_data',
                        data: c,
                    }
                }
            })
        }
        if (typeof params.title !== 'undefined') {
            result.title = params.title
        }
        if (typeof params.message !== 'undefined') {
            result.message = params.message
        }
        return result
    }

    FaceBook(shareConfig: FaceBookShareParams): Promise<ShareResult> {
        const shareType = `facebook_share_${shareConfig.type ?? 'messenger'}` as const

        return new Promise<ShareResult>((resolve, reject) => {
            if (this.platform === 'web') {
                window.open(`https://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(shareConfig.link ?? '')}`)
                resolve({
                    code: 0,
                    msg: '网页分享成功',
                })
            } else {
                // 调用端内
                EjoySDK.share(shareType, this.convertParams(shareConfig), {
                    onSuccess: (result) => {
                        if (result?.code === 0) {
                            resolve(result)
                        } else {
                            reject(result)
                        }
                    },
                    onError: (e) => {
                        reject({
                            code: -1,
                            msg: e.message,
                        })
                    },
                })
            }
        })
    }

    Twitter(shareConfig: TwitterShareParams): Promise<ShareResult> {
        return new Promise((resolve, reject: (reason: ShareResult) => void) => {
            if (this.platform === 'native') {
                if (__DEV__) {
                    logError('暂不支持分享到 twitter')
                }
                reject({
                    code: -1,
                    msg: '暂不支持分享到 twitter',
                })
            } else {
                window.open(`https://twitter.com/intent/tweet?text=${shareConfig.message ?? ''}&url=${shareConfig.link ?? ''}`)
                resolve({
                    code: 0,
                    msg: '网页分享成功',
                })
            }
        })
    }

    QQ(shareConfig: QQShareParams): Promise<ShareResult> {
        const { title = '', message = '', medias = [], link = '', type = 'qzone' } = shareConfig
        const shareType = `qq_share_${type}` as const

        return new Promise<ShareResult>((resolve, reject) => {
            if (this.platform === 'web') {
                let url = ''
                if (shareType === 'qq_share_qzone') {
                    url = `https://sns.qzone.qq.com/cgi-bin/qzshare/cgi_qzshare_onekey?url=${encodeURIComponent(
                        link,
                    )}&title=${encodeURIComponent(title)}&desc=${encodeURIComponent(message)}&pics=${medias
                        .map(encodeURIComponent)
                        .join('|')}`
                } else {
                    url = `https://connect.qq.com/widget/shareqq/index.html?url=${encodeURIComponent(link)}&title=${encodeURIComponent(
                        title,
                    )}&desc=${encodeURIComponent(message)}&pics=${medias.map(encodeURIComponent).pop() ?? ''}`
                }
                window.open(url)
                resolve({
                    code: 0,
                    msg: '网页分享成功',
                })
            } else {
                // 调用端内
                EjoySDK.share(shareType, this.convertParams(shareConfig), {
                    onSuccess: (result) => {
                        if (result?.code === 0) {
                            resolve(result)
                        } else {
                            reject(result)
                        }
                    },
                    onError: (e) => {
                        reject({
                            code: -1,
                            msg: e.message,
                        })
                    },
                })
            }
        })
    }

    Sina(shareConfig: SinaShareParams): Promise<ShareResult> {
        return new Promise<ShareResult>((resolve, reject) => {
            if (this.platform === 'web') {
                const { link = '', title = '', medias = [] } = shareConfig
                window.open(
                    `https://service.weibo.com/share/share.php?url=${encodeURIComponent(link)}&title=${encodeURIComponent(title)}&pic=${
                        medias.map(encodeURIComponent).pop() ?? ''
                    }`,
                )
                resolve({
                    code: 0,
                    msg: '网页分享成功',
                })
            } else {
                // 调用端内
                const shareType = `sina_share_${shareConfig.type ?? 'weibo'}` as const

                EjoySDK.share(shareType, this.convertParams(shareConfig), {
                    onSuccess: (result) => {
                        if (result?.code === 0) {
                            resolve(result)
                        } else {
                            reject(result)
                        }
                    },
                    onError: (e) => {
                        reject({
                            code: -1,
                            msg: e.message,
                        })
                    },
                })
            }
        })
    }

    // web类型的网页分享没有成功分享和取消分享的回调，应用设计中不要过分依赖此回调！
    WeChat(shareConfig: WechatShareParams): Promise<ShareResult> {
        return new Promise<ShareResult>((resolve, reject) => {
            if (this.platform === 'web') {
                // 微信分享只有微信内置 webview 才有效
                if (/MicroMessenger/i.test(navigator.userAgent)) {
                    if (!this.messager) {
                        if (__DEV__) {
                            logError('[wxConfig] 尚未配置，请参阅文档在初始化时配置')
                        }
                    } else {
                        this.messager.emit('wechatShare', () => {
                            const { link = '', title = '', message, medias = [] } = shareConfig
                            const shareData = {
                                title,
                                desc: message,
                                link, // 分享链接，该链接域名或路径必须与当前页面对应的公众号JS安全域名一致
                                imgUrl: medias[0] ?? '',
                            }

                            this.wx.updateAppMessageShareData(shareData)
                            this.wx.updateTimelineShareData(shareData)

                            resolve({
                                code: 0,
                                msg: '分享设置成功',
                            })
                        })
                    }
                } else {
                    reject({
                        code: -2,
                        msg: '请在微信浏览器打开',
                    })
                }
            } else {
                // 调用端内
                const shareType = `wechat_share_${shareConfig.type ?? 'messenger'}` as const

                EjoySDK.share(shareType, this.convertParams(shareConfig), {
                    onSuccess: (result) => {
                        if (result?.code === 0) {
                            resolve(result)
                        } else {
                            reject(result)
                        }
                    },
                    onError: (e) => {
                        reject({
                            code: -1,
                            msg: e.message,
                        })
                    },
                })
            }
        })
    }
}

export { SocialShare }
