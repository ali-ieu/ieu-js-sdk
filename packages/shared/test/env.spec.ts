import { Env } from '../src/env'

describe('Env', () => {
    it('it throws error without setting Host or setting to falsy value', () => {
        const env = new Env({ host: '' })
        expect(() => env.getHost()).toThrowErrorMatchingSnapshot()
    })
    it('it could change host by setHost method', () => {
        const env = new Env({ host: 'https://a.com' })
        expect(env.getHost()).toBe('https://a.com')
    })
    it('it could automatically complete protocol', () => {
        const env = new Env({ host: 'a.com' })
        expect(env.getHost()).toBe('https://a.com')
    })
})
