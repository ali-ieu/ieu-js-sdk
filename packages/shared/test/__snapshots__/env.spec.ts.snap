// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Env it throws error without setting Host or setting to falsy value 1`] = `"[host] 尚未初始化，请按照文档指示，设置一个明确的 Host 值!"`;
